components/gitpod-db/go/oidc_client_config.go

```diff
@@ -29,7 +30,31 @@ func (c *OIDCClientConfig) TableName() string {
 	return "d_b_oidc_client_config"
 }
 
+// It feels wrong to have to define re-define all of these fields.
+// However, I could not find a Go library which would include json annotations on the structs to guarantee the fields
+// will remain consistent over time (and resilient to rename). If we find one, we can change this.
 type OIDCSpec struct {
+	OAuth2   OAuth2Config   `json:"oauth2"`
+	Verifier VerifierConfig `json:"verifier"`
+}
+
+type OAuth2Config struct {
+	// ClientID is the application's ID.
+	ClientID string `json:"clientId"`
+
+	// ClientSecret is the application's secret.
+	ClientSecret string `json:"clientSecret"`
+
+	// RedirectURL is the URL to redirect users going through
+	// the OAuth flow, after the resource owner's URLs.
+	RedirectURL string `json:"redirectUrl"`
+
+	// Scope specifies optional requested permissions.
+	Scopes []string `json:"scopes"`
+}
+
+type VerifierConfig struct {
+	ClientID string `json:"clientId"`
```

Comment:
The separation between attributes for the verifier (part of OIDC process) and the auth flow seems not suitable when it comes to persistence, right? Duplication isn't needed at least. The verifier (from `go-oidc`) is a vehicle that requires some configuration which is not disjunct from the rest.

Let's make it a flat list of attributes. It would also simplify mapping like `toDBSpec`.